volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,505 @@
+        """Given a namespace ID it returns a dictionary mapping each network
+        interface index (ifindex) to its network interface name:
+
+        Args:
+            netns_id: The network namespace ID
+
+        Returns:
+            netdevices_map: Mapping network interface index (ifindex) to network
+                            interface name
+        """
+        netdevices_map = {}
+        nethead = self._vmlinux.object_from_symbol(symbol_name="net_namespace_list")
+        net_symname = self._vmlinux.symbol_table_name + constants.BANG + "net"
+        for net in nethead.to_list(net_symname, "list"):
+            net_device_symname = self._vmlinux.symbol_table_name + constants.BANG + "net_device"
+            for net_dev in net.dev_base_head.to_list(net_device_symname, "dev_list"):
+                if net.get_inode() != netns_id:
+                    continue
+                dev_name = utility.array_to_string(net_dev.name)
+                netdevices_map[net_dev.ifindex] = dev_name
+        return netdevices_map
+
+    def process_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str], Dict]:
+        """Takes a kernel generic `sock` object and processes it with its respective socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns a tuple with:
+            sock: The respective kernel's *_sock object for that socket family
+            sock_stat: A tuple with the source, destination and state strings.
+            extended: A dictionary with key/value extended information.
+        """
+        family = sock.family
+        extended = {}
+        sock_handler = self._sock_family_handlers.get(family)
+        if sock_handler:
+            try:
+                unix_sock, sock_stat = sock_handler(sock)
+                self._update_extended_socket_filters_info(sock, extended)
+
+                return unix_sock, sock_stat, extended
+            except exceptions.SymbolError as e:
+                # Cannot finds the *_sock type in the symbols
+                vollog.log(constants.LOGLEVEL_V, "Error processing socket family '%s': %s", family, e)
+        else:
+            vollog.log(constants.LOGLEVEL_V, "Unsupported family '%s'", family)
+
+        # Even if the sock family is not supported, or the required types
+        # are not present in the symbols, we can still show some general
+        # information about the socket that may be helpful.
+        saddr_tag = daddr_tag = state = "?"
+
+        sock_stat = saddr_tag, daddr_tag, state
+
+        return sock, sock_stat, extended
+
+    def _update_extended_socket_filters_info(self, sock: objects.Pointer, extended: dict) -> None:
+        """Get infomation from the socket and reuseport filters
+
+        Args:
+            sock: The kernel sock (sk) struct
+            extended: Dictionary to store extended information
+        """
+        if sock.has_member("sk_filter") and sock.sk_filter:
```

Comment:
@ikelos sk_filter here on line 109 and sk_reuseport_cb on 114 are pointers. Are the current checks ok? "and sock.sk_reuseport_cb" or does that need to be a in try/catch looking for InvalidAdressException?

Reply:
@ikelos bump on this comment

Reply:
Has member will return true as long as the structure is supposed to be present, I *think* you're ok with the pointer until you try to access any of its members, so this would do a test for 0 against the pointer...